### PR TITLE
RSWEB-7529: pattern(announcement): added header and hover state if link

### PR DESCRIPTION
**Pull request by runrog**
Needs design/code review. Added a header option and wrapped linked announcements in a custom a tag.
screen shot 2016-10-13 at 1 49 52 pm

### CODE REVIEW

File: styleguide/_themes/derek/scss/patterns/announcements.scss [hunk 1]
```diff
@@ -49,4 +57,8 @@
   .announcement-message {
     width: $full-width;
   }
+
+  .announcement-head {
+    text-align: center;
```
Comment (WidgetsBurritos) [inline styleguide/_themes/derek/scss/patterns/announcements.scss]: Any thing we do on the mobile version, needs to be added to the subpanel version as well.  I realize announcements aren't currently working on zoolander inside the subpanels page, but basically all that needs to happen is your mobile code should also be placed here:

https://github.com/rackerlabs/zoolander/blob/master/styleguide/_themes/derek/scss/patterns/panels.scss#L14

**Comment by runrog** [comment]
@amberRrucker design review por favor

**Comment by amberRrucker** [comment]
I was thinking more that the box would get an overlay or something when hovered over if there is a link. We also may want to still have space for the link text since mobile wont have a hover. Run it by mike with the text highlight and see what he thinks.

**Comment by runrog** [comment]
@amberRrucker I initially wanted to do a bg hover but it gets tricky since an announcment pattern that is a link will be getting wrapped in an 'a' tag and non links won't. a dynamic hover class would probly need to be added on the pattern container if it's a link. I'm going to push up another commit soon with an example (as requested by Stinemetze)

**Comment by WidgetsBurritos** [comment]
+1 on 19d014cb946604b0c99a1f6ac7d5cebe4df72f99

**Comment by amberRrucker** [comment]
Let's not have the paragraph change color on hover. It's a description, not a link so let's just give the whole area the gray hover state and make sure there is only a hover state for ones that have a link. So add a text link to one of them as the linked example.

**Comment by WidgetsBurritos** [comment]
👍  on 4179b2ed0e0bfc1d1e449e9eeb502f191f98af48

**Comment by amberRrucker** [comment]
The text is still highlighting on hover in the standard announcement examples.
Don't worry about showing them in the splits as those examples get shown on the subpanels page.
Please add link text to the linked version. Mobile won't have the hover so we need another visual cue that it's linking.

**Comment by WidgetsBurritos** [comment]
Also, you may want to do something about the focus state.  That looks funky:

image

File: styleguide/_themes/derek/scss/patterns/announcements.scss [hunk 2]
```diff
@@ -18,9 +18,25 @@
   }
 }
 
+.announcement-link {
+  line-height: inherit;
+  text-decoration: none;
+
+  &:hover {
+    text-decoration: none;
```
Comment (WidgetsBurritos) [inline styleguide/_themes/derek/scss/patterns/announcements.scss]: add `&:focus` here

File: styleguide/derek/_partials/cards/announcements-link.ejs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  'promotion' : 'fa-money'
+};
+%>
+<a href="#" class="announcement-link">
+  <div class="announcement-wrapper announcement-<%- type %> fade in row-eq-height announcement-link-hover">
+    <div class="announcement-icon flex-center-all"><i class="fa <%- icons[type] %>" aria-hidden="true"></i></div>
+    <div class="announcement-message">
+      <h3 class="announcement-head">Header Here</h3>
+      <%- desc %>
+      <span href="#" class="announcement-linkText">View More</span>
```
Comment (WidgetsBurritos) [inline styleguide/derek/_partials/cards/announcements-link.ejs]: No `href` here.

File: styleguide/derek/_partials/cards/announcements-link.ejs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+if(!desc) { var desc = "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium."; }
+if(!type) { var type = 'general'; }
+var icons = {
+  'alert' : 'fa-exclamation-circle',
+  'general' : 'fa-bullhorn',
+  'newProduct' : 'fa-server',
+  'promotion' : 'fa-money'
+};
+%>
+<a href="#" class="announcement-link">
```
Comment (WidgetsBurritos) [inline styleguide/derek/_partials/cards/announcements-link.ejs]: can you set the href to be `javascript: void(0)` instead?

**Comment by runrog** [comment]
@WidgetsBurritos done sir

**Comment by amberRrucker** [comment]
+1

**Comment by WidgetsBurritos** [comment]
Adding `HOLD` until www counterpart is complete.

**Pull request merged**